src/components/SourceCodeEditor.tsx

```diff
@@ -99,12 +92,16 @@ const SourceCodeEditor = (props: ISourceCodeEditor) => {
   };
 
   const debounced = useDebouncedCallback((value: string) => {
-    // if editor is in read only mode file can be still uploaded.
-    if (props.mode !== CodeEditorMode.FREE_EDIT) {
+    if (props.mode == CodeEditorMode.TWO_WAY_SYNC) {
```

Comment:
Can we not use `===` here?

Reply:
did you mean `==` ? because that's what I'm using. If yes what should I use instead ?

Reply:
usually in TypeScript and JavaScript world, we prefer to use the triple equal `===` instead of the double equal `==`. the double equal can do some weird type transformation.

Reply:
https://rules.sonarsource.com/typescript/RSPEC-1440

Reply:
Yep it's what @apupier said